src/Swift.Runtime/src/Swift/AnyType.cs

```diff
@@ -12,46 +13,49 @@ namespace Swift;
 /// </summary>
 public struct AnyType : ISwiftObject
 {
-    private SwiftHandle _payload = SwiftHandle.Zero;
+    private SwiftSafeHandle<AnyType> _payload = SwiftSafeHandle<AnyType>.Zero;
     static TypeMetadata ISwiftObject.GetTypeMetadata()
     {
-        return TypeMetadata.Zero;
+        throw new InvalidOperationException("Cannot get type metadata for AnyType");
     }
     public AnyType(SwiftHandle payload)
     {
-        _payload = payload;
+        _payload = new SwiftSafeHandle<AnyType>(payload);
     }
-    public SwiftHandle Payload => _payload;
+    public SwiftSafeHandle<AnyType> Payload => _payload;
 
     /// <summary>
     /// Creates a new SwiftOptional from a Swift payload
     /// </summary>
-    static ISwiftObject ISwiftObject.NewFromPayload(SwiftHandle payload)
+    static ISwiftObject ISwiftObject.NewFromPayload(IntPtr payload)
     {
         return new AnyType(payload);
     }
 
     /// <summary>
     /// Marshals this object to a Swift destination
     /// </summary>
-    /// <param name="swiftDest"></param>
+    /// <param name="swiftDestSpan"></param>
     /// <returns></returns>
-    IntPtr ISwiftObject.MarshalToSwift(IntPtr swiftDest)
+    void ISwiftObject.MarshalToSwift(Span<byte> swiftDestSpan)
     {
         var metadata = SwiftObjectHelper<AnyType>.GetTypeMetadata();
+        Debug.Assert((int)metadata.Size == swiftDestSpan.Length, $"Span size does not match type size, Expected: {(int)metadata.Size}, Actual: {swiftDestSpan.Length}");
```

Comment:
Seems like this assert should be after we check for valid metadata. It might be worth considering not having an `IsValid` on the this type. Why can't we just throw in the `SwiftObjectHelper` and avoid the check?

Reply:
I agree. AnyType is currently used only as a placeholder to allow the bindings to compile successfully. I've updated it to throw an exception if marshalling is attempted.